build.sh

```diff
@@ -50,6 +51,62 @@ check_managed_prereqs()
     fi
 }
 
+download_file()
+{
+    which curl wget > /dev/null 2> /dev/null
+    if [ $? -ne 0 -a $? -ne 1 ]; then
+        echo "cURL or wget is required to build corert."
+        exit 1
+    fi
+    echo "Downloading... $2"
+
+    # curl has HTTPS CA trust-issues less often than wget, so lets try that first.
+    which curl > /dev/null 2> /dev/null
+    if [ $? -ne 0 ]; then
+       wget -q -O $1 $2
+    else
+       curl -sSL --create-dirs -o $1 $2
+    fi
+    if [ $? -ne 0 ]; then
+        echo "Failed to download into $1 from $2."
+        exit 1
+    fi
+}
+
+install_dotnet_cli()
+{
+    echo "Installing the dotnet/cli..."
+    local __tools_dir=${__scriptpath}/bin/tools
+    local __cli_dir=${__tools_dir}/cli
+    
+    if [ ${__InstallCli} -ne 0 ]; then
+        rm -rf "${__cli_dir}"
+        if [ $? -ne 0 ]; then
+            echo "Could not remove existing CLI dir."
+            exit 1
+        fi
+    fi
+    if [ ! -d "${__cli_dir}" ]; then
+        mkdir -p "${__cli_dir}"
+    fi
+    if [ ! -f "${__cli_dir}/bin/dotnet" ]; then
+        local __build_os_lowercase=$(echo "${__BuildOS}" | tr '[:upper:]' '[:lower:]')
+        local __build_arch_lowercase=$(echo "${__BuildArch}" | tr '[:upper:]' '[:lower:]')
+        local __cli_tarball=dotnet-${__build_os_lowercase}-${__build_arch_lowercase}.latest.tar.gz
+        local __cli_tarball_path=${__tools_dir}/${__cli_tarball}
+        download_file ${__cli_tarball_path} "https://dotnetcli.blob.core.windows.net/dotnet/dev/Binaries/Latest/${__cli_tarball}"
+        tar -xzf ${__cli_tarball_path} -C ${__cli_dir}
+        export DOTNET_HOME=${__cli_dir}
+        if [ -n ${HOME:+1} ]; then
+            export HOME=${__tools_dir}
```

Comment:
I dislike doing this, but CLI's `dotnet restore` really wants HOME to be defined.

https://github.com/dotnet/cli/blob/5f5e3ad74c0c1de7071ba1309dca2ea289691163/scripts/ci_build.sh#L24